examples/gno.land/p/gov/dao/members.gno

```diff
@@ -0,0 +1,35 @@
+package dao
+
+import (
+	"std"
+)
+
+// MemberStore defines the member storage abstraction
+type MemberStore interface {
```

Comment:
Deserves Its own package.

A DAO is not necessarily based on members. It could be composed of other DAOs, among other things.

Let's create a memberstore package for DAOs that require a memberstore, as well as other non-DAO entities that need a memberstore.

Reply:
I've moved it out into a separate package:

[2f337a6](https://github.com/gnolang/gno/pull/2581/commits/2f337a665c683b476a39e782f4e8f8e409911606)

Reply:
Can you move these two interfaces in p/demo/membstore,so that Dao.membstore does not depend on gov//dao?

Reply:
>so that Dao.membstore does not depend on gov//dao

What do you mean by this?

The `dao.MemberStore` defines the interface that is now implemented in `p/demo/membstore`

Reply:
I'm not convinced we should be moving the memberstore interface outside `p/gov/dao`, since this should be a shared interface that implementations fulfill

Reply:
Until we have a way to to make anything addressable, including gno object, we should remove the member store from p/demo/dao.

A Dao isn’t necessarily about addressable users. And if for instance, we want to make interchain DAOs, we definitely don’t care at all about their members, we just do care about being able to propose things, and read proposals’ state.

Reply:
> make interchain DAOs

I had to pause for a second to grasp this concept. Powerful stuff

I've kept this commit stashed in case I got clarity on it, which I have now (`dao.DAO` is not the defactor API for DAOs yet, but a suggestion that should be moved out of `p/gov/dao`)

Moved:
[06fe4fe](https://github.com/gnolang/gno/pull/2581/commits/06fe4fe33428b4cda3f91b1bcd03175165f9ae6b)